addon.py

```
@@ -419,21 +510,44 @@ def get_viewport_screenshot(self, max_size=800, filepath=None, format="png"):
+                "mathutils": mathutils,
+                "context": bpy.context
+            }
 
-            # Capture stdout during execution, and return it as result
+            # Capture stdout during execution
             capture_buffer = io.StringIO()
             with redirect_stdout(capture_buffer):
-                exec(code, namespace)
+                # Ensure we are in a valid state for common operators
+                if bpy.context.screen:
+                    exec(code, namespace)
```

Ruff (0.15.6)

[error] Use of `exec` detected

(S102)